src/main/java/de/tum/in/www1/artemis/web/rest/ApollonConversionResource.java

```diff
@@ -0,0 +1,52 @@
+package de.tum.in.www1.artemis.web.rest;
+
+import java.io.InputStream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import de.tum.in.www1.artemis.service.connectors.apollon.ApollonConversionService;
+import de.tum.in.www1.artemis.service.connectors.apollon.dto.ApollonModelDTO;
+
+/**
+ * REST controller for managing ApollonDiagram.
+ */
+@RestController
+@RequestMapping("/api")
+@Profile("apollon")
+public class ApollonConversionResource {
+
+    private final Logger log = LoggerFactory.getLogger(ApollonConversionResource.class);
+
+    private final ApollonConversionService apollonConversionService;
+
+    public ApollonConversionResource(ApollonConversionService apollonConversionService) {
+        this.apollonConversionService = apollonConversionService;
+    }
+
+    /**
+     * Converts given model to pdf
+     * @param request the model for conversion
+     * @return input stream for conversion
+     */
+    @PostMapping("/apollon-convert/pdf")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
```

Comment:
maybe this is due to my lack of knowledge, but isn't `@PreAuthorize("hasRole('USER')")`  enough for that?